src/actions/on-message.ts

```diff
@@ -1,25 +1,18 @@
 import { Bot } from "../services/bot/bot";
-import { Message, MessageType } from "../models/message";
-import { onMentions } from "./on-mentions";
+import { Message, MessageType, isMessage } from "../models/message";
+import { isError } from "../models/error";
 import { onGratitude } from "./on-gratitude";
+import { onError } from "./on-error";
 
-const isNotMessage = (data) => {
-  return data.type !== 'message'
-    || data.subtype === 'bot_message'
-    || data.subtype === 'message_changed'
-    || data.bot_id !== undefined
-}
+export const onMessage = (bot: Bot, data: any) => {
+  if (isError(data)) onError(bot, data)
 
-export const onMessage = (bot: Bot, data) => {
-  if (isNotMessage(data)) return;
-
-  const message: Message = new Message(data);
-  switch (message.type) {
-    case MessageType.Mention:
-      onMentions(bot, message);
-      break;
-    case MessageType.Gratitude:
-      onGratitude(bot, message);
-      break;
+  if (isMessage(data)) {
+    const message: Message = new Message(data);
+    switch (message.type) {
+      case MessageType.Gratitude:
+        onGratitude(bot, message);
+        break;
+    }
```

Comment:
Maybe we can move it on a map handler?
```
const messageHandler = {
  [MessageType.Gratitude]: () => { onGratitude(bot, message); }
}

messageHandler[message.type]();
```

Reply:
And in case of "default" it does nothing and that's it?

Reply:
We should have to check that we have the defition, or do nothing if the type doesn't exist, but you are not handling a default value right now.

Maybe we can do something like this
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html

Or, we use the switch as you are doing right now :p

Reply:
Check the changes